### PR TITLE
WIP: VAMDC-cdms interface

### PR DESCRIPTION
**Pull request by keflavich**
See #618.  The goal of this PR is to include an API similar to the Splatalogue
API for accessing information from the CDMS database.

cc @vilhelmp

### CODE REVIEW

**Comment by keflavich** [comment]
I now use this regularly in some code.  @vilhelmp, any chance you could review it?  Otherwise, @bispocz, any objections to merging this in its current state?  It is a bit sparse on functionality, but it does something.

**Comment by keflavich** [comment]
Wait, nevermind - can't merge this until I add some tests!  And it requires `vamdclib`.

**Comment by bsipocz** [comment]
Since there are quite a few modules that are far from being complete, I think it's fine to merge this. 
Although a rebase, to get rid of the irrelevant merge commits, and some tests would be nice.

**Comment by bsipocz** [comment]
There isn't much info about the install and deps on the docs pages of vamdclib, so I'm just blindly making suggestions here:
We could either try to build a conda package for it, or install it separately in .travis.yml. Former is better if it takes a long time to install, and/or there are other packages that may use it, latter seems simpler.

**Comment by vilhelmp** [comment]
Would love to test it out. I just switched to using Anaconda Python though. Have to figure out how to test a separate build. Virtualenv I guess?
@keflavich @bsipocz

**Comment by keflavich** [comment]
you can make a separate conda environment, but I usually just use `python setup.py develop` and switch between branches to test new features

**Comment by keflavich** [comment]
I'm going to link `.travis.yml` against my personal fork of `vamdclib` because I've incorporated `package-template` into it.  Otherwise, my branch should match upstream, and this puts the burden on me to keep it up-to-date.  I'll try to convince the vamdc maintainer to incorporate package-template though...

**Comment by bsipocz** [comment]
@keflavich - I wonder whether you have the ``vamdclib`` from upstream? It looks like that cpe/vamdclib has only one fork, yours?

**Comment by keflavich** [comment]
@bsipocz this is a confusion situation.  There are TWO identical upstreams: `cpe/vamdclib` and `VAMDC/vamdclib`.  I have forks of both... I think they were created independently but should not be independent.  I filed a request with github support to try to fix the situation, but I'm not sure there's anything they can do about it.  In any case, I just submitted a [PR](https://github.com/VAMDC/vamdclib/pull/1) to incorporate `package-template`

**Comment by bsipocz** [comment]
Sounds good.

**Comment by keflavich** [comment]
It seems to be nontrivial to create a non-remote-dependent vamdclib request.  I may elect for remote-only tests for this module.  It will result in more likely breakage, but I'm not sure there's anything to be done about that.  It's a fairly niche module anyway.

**Comment by bsipocz** [comment]
Sure. In that case I would suggest to add a second line in travis where we run the tests that runs the remote for only for this module.

**Comment by keflavich** [comment]
@bsipocz do you know what's going on with https://travis-ci.org/astropy/astroquery/jobs/169277855#L1303?  I don't understand why it's complaining about not having a VAMDC API page:
```
/home/travis/build/astropy/astroquery/docs/vamdc/vamdc.rst:49: WARNING: toctree references unknown document u'api/astroquery.vamdc.Vamdc'
```
since that should be generated.

**Comment by bsipocz** [comment]
@keflavich - This is weird, locally there are no warnings, but I've seen similar warnings before. Let me dig a bit deeper.

**Comment by bsipocz** [comment]
I've restarted travis to see whether it was just a fluke. Also could you please rebase to get rid of the merge commits?

File: astroquery/vamdc/core.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+        # example use of specmodel; return to this later...
+        #Q = self.specmodel.calculate_partitionfunction(result.data['States'],
+        #                                  temperature=tex)[species_id]
+
+
+try:
+    Vamdc = VamdcClass()
+except ImportError:
+    log.warn("vamdclib could not be imported; the vamdc astroquery module will not work")
```
Comment (bsipocz) [inline astroquery/vamdc/core.py]: pls use log.warning as warn is deprecated

**Comment by bsipocz** [comment]
@keflavich - so the sphinx warning indeed went away, and you could get rid of the remaining failures with the warn-->warning change.

**Comment by keflavich** [comment]
@bsipocz I can rebase, but is there any particular reason to?  I don't mind having merge commits around so much.

Thanks for catching `warn` vs `warning`... I knew one was deprecated but can never remember which.

**Comment by bsipocz** [comment]
No strong reason, I usually prefer to have a cleaner branch structure without the merges as it's slightly easier to see what going on.

**Comment by keflavich** [comment]
OK, rebased and I think it's ready to merge.  Anything to add @bsipocz?

**Comment by bsipocz** [comment]
@keflavich - Sure, let's get this merged.

**Comment by vilhelmp** [comment]
I quickly made a small function to query CDMS through their web interface. https://github.com/vilhelmp/cdmspy
Don't know if there's any use in getting it into astroquery?

**Comment by keflavich** [comment]
@vilhelmp yeah, even though it's somewhat redundant with vamdc, I think having a direct interface to their web page is valuable.  vamdclib doesn't fall within the astropy ecosystem at all, and it doesn't follow all the 'normal' python conventions, so I don't know how much we can rely on it long term.  Please open a PR!

**Pull request merged**